kpi/tests/test_usage_calculator.py

```diff
@@ -0,0 +1,220 @@
+            owner=owner,
+            asset_type='survey',
+        )
+
+        self.asset.deploy(backend='mock', active=True)
+        self.asset.save()
+
+        self.asset.deployment.set_namespace(self.URL_NAMESPACE)
+        self.submission_list_url = reverse(
+            self._get_endpoint('submission-list'),
+            kwargs={'format': 'json', 'parent_lookup_asset': self.asset.uid},
+        )
+        self._deployment = self.asset.deployment
+
+
+    def add_nlp_trackers(self):
+        """
+        Add nlp data to an asset
+        """
+        # this month
+        today = timezone.now().date()
+        counter_1 = {
+            'google_asr_seconds': 4586,
+            'google_mt_characters': 5473,
+        }
+        NLPUsageCounter.objects.create(
+            user_id=self.anotheruser.id,
+            asset_id=self.asset.id,
+            date=today,
+            counters=counter_1,
+            total_asr_seconds=counter_1['google_asr_seconds'],
+            total_mt_characters=counter_1['google_mt_characters'],
+        )
+
+        # last month
+        last_month = today - relativedelta(months=1)
+        counter_2 = {
+            'google_asr_seconds': 142,
+            'google_mt_characters': 1253,
+        }
+        NLPUsageCounter.objects.create(
+            user_id=self.anotheruser.id,
+            asset_id=self.asset.id,
+            date=last_month,
+            counters=counter_2,
+            total_asr_seconds=counter_2['google_asr_seconds'],
+            total_mt_characters=counter_2['google_mt_characters'],
+        )
+
+    def add_submissions(self, count=2):
+        """
+        Add one or more submissions to an asset (TWO by default)
+        """
+        submissions = []
+        v_uid = self.asset.latest_deployed_version.uid
+
+        for x in range(count):
+            submission = {
+                '__version__': v_uid,
+                'q1': 'audio_conversion_test_clip.3gp',
+                'q2': 'audio_conversion_test_image.jpg',
+                '_uuid': str(uuid.uuid4()),
+                '_attachments': [
+                    {
+                        'id': self.attachment_id,
+                        'download_url': 'http://testserver/anotheruser/audio_conversion_test_clip.3gp',
+                        'filename': 'anotheruser/audio_conversion_test_clip.3gp',
+                        'mimetype': 'video/3gpp',
+                    },
+                    {
+                        'id': self.attachment_id + 1,
+                        'download_url': 'http://testserver/anotheruser/audio_conversion_test_image.jpg',
+                        'filename': 'anotheruser/audio_conversion_test_image.jpg',
+                        'mimetype': 'image/jpeg',
+                    },
+                ],
+                '_submitted_by': 'anotheruser',
+            }
+            # increment the attachment ID for each attachment created
+            self.attachment_id = self.attachment_id + 2
```

Comment:
We can get rid of `self.attachment_id` since `_attachments[index]['id']` is set directly in `mock_submissions()`. 
If we use in somewhere in the tests, we need to use something else to get the id of the attachment (e.g. `submission[0]['_attachments'][idx]`)

Reply:
Ok, also removed the self.counter property